src/features/accuse/index.tsx

```diff
@@ -128,9 +129,9 @@ export function Accuse() {
         <Button
           mt={12}
           px={6}
-          disabled={isError || !isEnabled}
+          disabled={!isEnabled || mayFail || signatures.length !== archaeologistPrivateKeys.length}
           onClick={handleAccuse}
-          isLoading={isLoading || isAccusing}
+          isLoading={!isError && (isLoading || isAccusing)}
```

Comment:
Unset loading flag if actual transaction reverts (note `isError` here is from `useWaitTransaction`. `isError` from `usePrepareContractWrite` is aliased to `mayFail`)